opensfm/src/CMakeLists.txt

```diff
@@ -54,11 +54,21 @@ endif()
 
 find_package(LAPACK)
 find_package(SuiteSparse)
-find_package(Eigen REQUIRED)
-find_package(Ceres REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(Ceres)
 find_package(Gflags REQUIRED)
 find_package(Glog REQUIRED)
 
+# Ceres2 exposes Ceres::ceres target.
+# Ceres1 exposes just ceres.
+# - if there's no such target, cmake will convert it into -lceres
+# and the linker will fail
+if(Ceres_ceres_FOUND)
+    set(CERES_LIBRARIES Ceres::ceres)
```

Comment:
I'm still using (but also explicitly defining the `CERES_LIBRARIES` variable because one needs to `target_link_libraries` different targets, conditionally on the version of ceres

Reply:
Hello! I'm not sure if it's possible: if we keep `FindCeres.cmake`, then `find_package(ceres ...)` a few lines above would probably always encounter the in-tree `FindCeres.cmake` before it checks the other directories in `CMAKE_PREFIX_PATH`. I'll try to research that